collect_app/src/test/java/org/odk/collect/android/http/TestableCollectServerClient.java

```diff
@@ -1,6 +1,8 @@
 package org.odk.collect.android.http;
 
 import org.odk.collect.android.http.mock.DaggerMockHttpComponent;
+import org.odk.collect.android.http.mock.MockHttpClientConnection;
+import org.odk.collect.android.http.mock.MockHttpInterfaceModule;
 
 public final class TestableCollectServerClient extends CollectServerClient {
```

Comment:
Hey @xsteelej since we have a testable version of this here are we gonna be injecting the `CollectServerClient` across the app instead of using `getInstance`? Doing so would allow us to use this mocked version in extensive instrumentation tests. What do you think?

Reply:
Thanks @jd-alexander, you're right it definitely should be. This is where my knowledge of Dagger gets a little cloudy so any pointers would be handy, but would you have the Collect class hold on to the CollectServerClient instance and then the unit tests could swap it out for a mocked version?

There seems to be similarities between your SMS PR and the PR that @shobhitagarwal1612 is working on: https://github.com/opendatakit/collect/pull/1871 in that they both have the need to swap out instances for testing. I need to be consistent and follow the same approach.

Reply:
Nah, I would just let `Dagger` hold onto it and make it a singleton. You are doing good so far! I am learning how to use Dagger as I go along as well because it's a pretty vast DI framework especially since Google took over.. Lots to learn!

Reply:
The CollectServerClient methods are called from other static methods, which is a bit of a shame as that makes it more tricky to remove the getInstance() and the static methods in CollectServerClient.

Reply:
Okay no problem. Later today I am gonna take a look and try to create a `CollectServerClient` without it.